src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -148,6 +148,7 @@
     public static final String GIT_COMMIT = "GIT_COMMIT";
     public static final String GIT_PREVIOUS_COMMIT = "GIT_PREVIOUS_COMMIT";
     public static final String GIT_PREVIOUS_SUCCESSFUL_COMMIT = "GIT_PREVIOUS_SUCCESSFUL_COMMIT";
+    public static final String GIT_CEILING_DIRECTORIES = "GIT_CEILING_DIRECTORIES";
```

Comment:
I'm not confident that this will have the desired behavior.  I believe you want to control the behavior of command line git and JGit.  This provides an environment variable to the processes that run on the agent as part of a job.  I am not sure if the value of the environment variable defined here will alter the behavior of command line git.

This is a very good place to write a test that shows the failure as described in the bug report, then see if this change resolves the failure in that test.  Before writing the failing automated test, you may also want to explore it interactively to see it fail from within Jenkins.
```suggestion
    public static final String GIT_CEILING_DIRECTORIES = "GIT_CEILING_DIRECTORIES";
```